fighthealthinsurance/urls.py

```diff
@@ -14,31 +14,110 @@
 
 from fighthealthinsurance import views
 from fighthealthinsurance import fax_views
 from fighthealthinsurance import staff_views
 from django.views.decorators.debug import sensitive_post_parameters
 import os
-
+from django.conf import settings
 
 def trigger_error(request: HttpRequest) -> HttpResponseBase:
     raise Exception("Test error")
 
 
+# Add security middleware
+class SecurityScanMiddleware:
+    def __init__(self, get_response: Callable[[HttpRequest], HttpResponseBase]):
+        self.get_response = get_response
+        self.security_patterns = [
+            r'(?i)(<|%3C)script',  # XSS attempts
+            r'(?i)(union\s+select|insert\s+into|drop\s+table|delete\s+from)',  # SQL injection attempts
+            r'(?i)(/\.\./|\.\./)',  # Path traversal attempts
+            r'(?i)(exec\s+|eval\s*\()',  # Code injection attempts
+            r'(?i)(file|system|phpinfo)\s*\(',  # PHP injection attempts
+        ]
+        self.patterns = [re.compile(pattern) for pattern in self.security_patterns]
+        self.request_history: Dict[str, List[float]] = {}
+        self.rate_limit = 30  # requests
+        self.time_window = 300  # 5 minutes in seconds
+        self.max_history_size = 10000  # Maximum number of IPs to track
+
+    def __call__(self, request: HttpRequest) -> HttpResponseBase:
+        # Skip security checks for test environment
+        if settings.TESTING:
+            return self.get_response(request)
+
+        # Check for security scan patterns
+        path = request.path
+        query = request.META.get('QUERY_STRING', '')
+        content = path + query
+        
+        # Get client IP with a default value and type checking
+        client_ip: str = request.META.get('REMOTE_ADDR', '')
+        if not client_ip:
+            return HttpResponseForbidden("Could not determine client IP")
+
+        # Skip rate limiting for authenticated staff/admin users
+        if request.user.is_authenticated and (request.user.is_staff or request.user.is_superuser):
+            return self.get_response(request)
+
+        # Rate limiting for sensitive endpoints
+        sensitive_patterns = ['login', 'logout', 'password_reset', 'rest_verify_email', 'admin', 'webhook']
+        if any(pattern in path.lower() for pattern in sensitive_patterns):
+            current_time = time.time()
+            
+            # Clean up old entries
+            self.request_history = {
+                ip: times for ip, times in self.request_history.items()
+                if times[-1] > current_time - self.time_window
+            }
+            
+            # Check rate limit
+            if client_ip in self.request_history:
+                times = self.request_history[client_ip]
+                if len(times) >= self.rate_limit:
+                    return HttpResponseForbidden("Rate limit exceeded")
+                times.append(current_time)
+            else:
+                self.request_history[client_ip] = [current_time]
```

Comment:
So as this is done it's going to be per Python process? That doesn't work super well with scaling but if the session are sticky I guess this might be ok.